test/functions/index.js

```diff
@@ -62,6 +62,81 @@ module.exports.deleteReferencesToMaster = integrify({
   },
 });
 
+module.exports.deleteReferencesWithMasterParam = integrify({
+  rule: 'DELETE_REFERENCES',
+  source: {
+    collection: 'master',
+  },
+  targets: [
+    {
+      collection: 'detail1',
+      foreignKey: 'masterId',
+    },
+    {
+      collection: 'somecoll/$masterId/detail2',
+      foreignKey: 'masterId',
```

Comment:
Seems strange to use a the variable $masterId which is not defined in the source path  - where is it coming from?

Reply:
The `masterId` is the id of the source collection. It would come from the path though. You think it would be easier to understand if we used `{...}` for path variables and `$...` for field variables?

Reply:
if it comes from the source path where is it defined, shouldnt `$masterId` in the target throw an error because the source  path doesnt define the variable?

Reply:
Is masterId a magic var created from 'Id' appended on collection name?

in the security rules syntax you would include the id in the source path if you wanted to reference it the rule:

`match /users/{uid} {` ...

Maybe we can do the same here?

```
source: {
    collection: 'master/{masterId}',
  },
  targets: [
    {
      collection: 'somecoll/$masterId/detail2',
       ...
    },
```

Reply:
```
const paramSwap = replaceReferenceWithFields(context.params, target.collection);
```
This line replaces the `$...` with the params, which masterId is part of. The next `replaceReferenceWithFields` replaces the `$..` with the fields. (Looking at it now the function name could be changed to some like `replaceReferenceWith(...)` to read better)

Reply:
Yeah, it has just been defined for us already and they chose `masterId` (line 34 in deleteReference.ts). We could use `collection: 'master/{masterId}'` to make it easier to understand too

Reply:
so its a secret magic var caleld masterId! Really confusing as docs dont state this and their example collection is called master so you might conclude it comes from the collection name. 

Ok would make a lot of sense to require it to be defined in the source path and respects the name give to it

```
source: {
    collection: 'users/{userId}',
  },
  targets: [
    {
      collection: 'somecoll/$userId/detail2',
       ...
    },
```